m4/check_enable_ubsan.m4

```diff
@@ -0,0 +1,18 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2023      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_ENABLE_UBSAN], [
+  AC_MSG_CHECKING([whether to enable UBSAN (Undefined Behaviour Sanitizer)])
+  AC_ARG_ENABLE([ubsan],
+                [AS_HELP_STRING([--enable-ubsan], [Enable undefined behaviour checks with UBSAN])])
+
+  AS_IF([test "${enable_ubsan}" = "yes"],
```

Comment:
you should run a test compile of a simple application to ensure that -fsanitize=undefined actually works with this compiler.